### DISCUSSION TITLE
Passing conditions to Netconf driver Get and GetConfig

### DISCUSSION THREAD

**Discussion by dau71** [scrapli/scrapligo#137] [Q&A]
Is it possible to pass conditions to the Get and GetConfig methods of the netconf driver?

**Comment by hellt** [comment]
What do you mean by that? You can pass filters

**Reply by dau71** [reply]
I want to be able to create a filter that results in  a similar rpc call to the below.  Filter on interfaces where the type is an onu.
`
<rpc xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="12">
  <get>
    <filter type="subtree">
      <if:interfaces-state xmlns:if="urn:ietf:params:xml:ns:yang:ietf-interfaces" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0">
        <if:interface>
          <if:name/>
          <if:type xmlns:adtn-xp="http://www.adtran.com/ns/yang/adtran-xpon">adtn-xp:xpon-onu</if:type>
          <adtn-xp:onu xmlns:adtn-xp="http://www.adtran.com/ns/yang/adtran-xpon"/>
        </if:interface>
      </if:interfaces-state>
    </filter>
  </get>
</rpc>
`

**Reply by carlmontanari** [reply]
yes, you can do this with filters like roman mentioned -- filter types and filters themselves are passed as options, you can see them [here](https://github.com/scrapli/scrapligo/blob/main/driver/opoptions/netconf.go). there are some tests [here](https://github.com/scrapli/scrapligo/blob/main/driver/opoptions/netconf_test.go) where you can see this in action. if you want to open a pr to add an example with this once you've got it setup that would be great!

**Reply by dau71** [reply]
Thanks, @carlmontanari and @hellt, my filters were wrong.
